src/components/store/ui/Product.tsx

```diff
@@ -0,0 +1,212 @@
+>((props, ref) => {
+  return (
+    <ProductPrimitive.Name
+      {...props}
+      ref={ref}
+      className={cn(
+        'text-4xl font-bold text-content-primary mb-4 font-theme-heading',
+        props.className
+      )}
+    >
+      {props.children}
+    </ProductPrimitive.Name>
+  );
+});
+
+ProductName.displayName = 'ProductName';
+
+export const ProductDescription = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.Description>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.Description>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.Description
+      {...props}
+      ref={ref}
+      className={cn(
+        'text-content-secondary text-base leading-relaxed mb-6',
+        props.className
+      )}
+    >
+      {props.children}
+    </ProductPrimitive.Description>
+  );
+});
+
+ProductDescription.displayName = 'ProductDescription';
+
+export const ProductPrice = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.Price>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.Price>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.Price
+      {...props}
+      ref={ref}
+      className={cn(
+        'text-3xl font-bold text-content-primary data-[discounted]:text-status-success',
+        props.className
+      )}
+    >
+      {props.children}
+    </ProductPrimitive.Price>
+  );
+});
+
+ProductPrice.displayName = 'ProductPrice';
+
+export const ProductCompareAtPrice = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.CompareAtPrice>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.CompareAtPrice>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.CompareAtPrice
+      {...props}
+      ref={ref}
+      className={cn(
+        'text-lg text-content-muted line-through hidden data-[discounted]:inline',
+        props.className
+      )}
+    >
+      {props.children}
+    </ProductPrimitive.CompareAtPrice>
+  );
+});
+
+ProductCompareAtPrice.displayName = 'ProductCompareAtPrice';
+
+export const ProductSlug = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.Slug>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.Slug>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.Slug
+      {...props}
+      ref={ref}
+      className={cn(
+        'text-content-secondary font-mono text-sm',
+        props.className
+      )}
+    >
+      {props.children}
+    </ProductPrimitive.Slug>
+  );
+});
+
+ProductSlug.displayName = 'ProductSlug';
+
+export const ProductRaw = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.Raw>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.Raw>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.Raw
+      {...props}
+      ref={ref}
+      className={cn('block', props.className)}
+    >
+      {props.children}
+    </ProductPrimitive.Raw>
+  );
+});
+
+ProductRaw.displayName = 'ProductRaw';
+
+export const ProductMediaGallery = React.forwardRef<
```

Comment:
why did you have to define it and not just 
`export const ProductMediaGallery = ProductPrimitive.ProductMediaGallery`?

Reply:
Cause ProductPrimitive.ProductMediaGallery has a default container <div>